File: java/classic/dynamic_programming/FibonacciSequence.java
package classic.dynamic_programming;

public class FibonacciSequence {
    public int[] solution(int num) {
        int result[] = new int[num];
        result[0] = 1;
        result[1] = 1;
        for (int i = 2; i < num; ++i) {
            result[i] = result[i - 1] + result[i - 2];
        }
        return result;
    }
}

File: py/classic/dynamic_programming/FibonacciSequence.py
#!/usr/bin/env python
# -*- coding:gbk -*-

from typing import List

class FibonacciSequence:
    def solution(self, num: int) -> List[int]:
        result = [1, 1]
        for _ in range(2, num):
            result.append(result[-1] + result[-2])
        return result
